lib/authz/backend/static/static_test.go

```diff
@@ -0,0 +1,120 @@
+package static
+
+import (
+	"context"
+	"tcplb/lib/core"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type DummyClientID string
+
+type DummyUpstream string
+
+func (u DummyUpstream) Name() string {
+	return string(u)
+}
+
+func TestAuthorizer(t *testing.T) {
+	alice := DummyClientID("alice")
+	bob := DummyClientID("bob")
+	cindy := DummyClientID("cindy")
+	dinesh := DummyClientID("dinesh")
+	eve := DummyClientID("eve")
+
+	alpha := NewGroup("alpha")
+	beta := NewGroup("beta")
+	admin := NewGroup("admin")
+
+	web := NewUGroup("web")
+	worker := NewUGroup("worker")
+
+	web1 := DummyUpstream("web1")
+	web2 := DummyUpstream("web2")
+	worker1 := DummyUpstream("worker1")
+	worker2 := DummyUpstream("worker2")
+
+	cfgZero := Config{}
+
+	cfgSmall := Config{
+		GroupsByClientID: map[core.ClientID][]Group{
+			alice:  []Group{admin},
+			bob:    []Group{beta, alpha},
+			cindy:  []Group{beta},
+			dinesh: []Group{alpha},
```

Comment:
nit: this can be simplified a bit
```suggestion
			alice:  {admin},
			bob:    {beta, alpha},
			cindy:  {beta},
			dinesh: {alpha},
```